CMakeLists.txt

```diff
@@ -5,11 +5,22 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 add_compile_options(
-    -Werror=incompatible-pointer-types
-    -Werror=implicit-function-declaration
+    #-Werror=incompatible-pointer-types
+    #-Werror=implicit-function-declaration
+    #-w
+    -g
     -mfpmath=387
 )
```

Comment:
Can we still use warning flags please. like -Wall and -Werror. Even if its not -Werror so that it doesn't block compilation, it would be better long term, such as explicit casting and signed / unsigned comparisons

Reply:
those two flags find a lot of typical c problems, this is why i used them. i fixed ~20 bugs using those flags.

Reply:
and about '-g': the correct way to enable debug symbols in cmake is `-DCMAKE_BUILD_TYPE=Debug` via command line or `set(CMAKE_BUILD_TYPE Debug)` at the beginning of the CMakeLists.txt

Reply:
I actually did not mean to submit this part of the change, I will revert it.